ilastik/applets/trainableDomainAdaptation/trainableDomainAdaptationGui.py

```diff
@@ -0,0 +1,227 @@
+import logging
+from functools import partial
+
+import sip
+from PyQt5.QtCore import Qt, QTimer, pyqtSlot
+from PyQt5.QtGui import QColor
+from PyQt5.QtWidgets import QAction, QColorDialog, QFileDialog, QHBoxLayout, QLabel, QMenu, QPushButton
+from volumina.api import AlphaModulatedLayer, LazyflowSource
+from volumina.colortables import default16_new
+
+from ilastik.applets.pixelClassification.pixelClassificationGui import PixelClassificationGui
+
+from ilastik.applets.pixelClassification.SuggestFeaturesDialog import SuggestFeaturesDialog
+from .modelStateControl import EnhancerModelStateControl
+
+logger = logging.getLogger(__name__)
+
+
+class TrainableDomainAdaptationGui(PixelClassificationGui):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._drawer = self._labelControlUi
+        self.liveNNPrediction = False
+        self.__cleanup_fns = []
+        self._init_channel_selector_ui()
+        self._init_nn_prediction_ui()
+
+        self.invalidatePredictionsTimer = QTimer()
+        self.invalidatePredictionsTimer.timeout.connect(self.updateNNPredictions)
+        self.tiktorchModel.registerListener(self._onModelStateChanged)
+
+        self.labelingDrawerUi.liveUpdateButton.toggled.connect(self.toggleLiveNNPrediction)
+
+    def _init_channel_selector_ui(self):
+        drawer = self._drawer
+
+        channel_selector = QPushButton()
+        self.channel_menu = QMenu(self)  # Must retain menus (in self) or else they get deleted.
+        channel_selector.setMenu(self.channel_menu)
+        channel_selector.clicked.connect(channel_selector.showMenu)
+
+        def populate_channel_menu(*args):
+            if sip.isdeleted(channel_selector):
```

Comment:
Is it possible to keep `channel_selector` in an instance attribute of `self` to prevent it from being deleted too, like `channel_menu`?

Reply:
phew again, more than a year ago this might have had a reason somehow? :D I tried without and ... it does seem to work without it, even without holding on to a reference to channel selector...

Reply:
Seems OK since `channel_selector` gets it's parent through adding it to a layout later.

Reply:
thx for checking